### ISSUE TITLE
Fix S4275: FP when using MVVM frameworks

### ISSUE THREAD

**Issue by alanwol** [SonarSource/sonar-dotnet#2140]
### Description

Rule S4275 gives a false positive when the setter is implemented by a member method taking in a reference such as the one implemented for the MVVM Light framework.

### Repro steps

```csharp
public double MyProp
{
     get => this._myProp;
     set
     {
          if (this.Set(ref this._myProp, value))
          {
                this._isDirty = true;
          }
     }
}
```

### Related information

SonarLint version: 4.7.0.3891
SonarQube c# plugin: 7.9.1 (build 7622) 
Visual Studio 15.8.2
SonarQube scanner 4.3.2
